python/mlx/optimizers/schedulers.py

```diff
@@ -84,3 +85,33 @@ def scheduler(step):
         return init * decay
 
     return scheduler
+
+
+def linear_warmup(
+    schedule_fn, length: int, finish: float, init: float = 0.00
+) -> Callable:
+    """
+    >>> lr_schedule = optim.cosine_with_warmup(100, 1e-1)
+    >>> optimizer = optim.Adam(learning_rate=lr_schedule)
+    >>> optimizer.learning_rate
+    array(0.0, dtype=float32)
+    >>> for _ in range(100): optimizer.update({}, {})
+    ...
+    >>> optimizer.learning_rate
+    array(0.1, dtype=float32)
+
+    :param schedule_fn: other schedule function, such as cosine_decay,step_decay, etc.
+    :param length: warmup length
+    :param init: start value (0 by default)
+    :param finish: final value
+    :return: callable takes a step and returns the schedulable
+    """
+
+    def schedule_step_fn(step):
+        if step <= length:
+            return step * ((finish - init) / length) + init
+        else:
+            offset_idx = step.item() - length
```

Comment:
Why call `step.item()` here? I think it's important to leave it as an `mx.array` and not eval the graph.

Reply:
Ok. I'll remove that and make it:

```python
step - length
```

 I wasn't sure if I could assume using an arithmetic operation on both length and step would work whether they were scalars or mx.arrays.

Reply:
Yup that will work fine! You can always do ops with one `mx.array` and a Python scalar. Doing the `item()` also works in most cases but it would cause issues with compilation so that's one reason to not have it.